components/utility/Scanner.tsx

```diff
@@ -0,0 +1,213 @@
+import React, { useState, useEffect, useRef } from "react";
+import { BrowserMultiFormatOneDReader, IScannerControls } from "@zxing/browser";
+
+type Device = {
+	deviceId: string;
+	label: string;
+};
+
+const ScanBarcode: React.FC = () => {
+	const controlsRef = useRef<IScannerControls | null>(null);
+	const [selectedDeviceId, setSelectedDeviceId] = useState("");
+	const [devices, setDevices] = useState<Array<Device>>([]);
+	const [name, setName] = useState("");
+	const [brand, setBrand] = useState("");
+
+	const getQRCodeReaderControls = async (selectedDeviceId: string) => {
+
+		try {
+			const codeReader = new BrowserMultiFormatOneDReader();
+			const previewElem = document.querySelector("video");
+			const videoElem = previewElem as HTMLVideoElement;
+
+			const BASE_URL =
+				"https://gb-be.de/";
+
+			const controls =
+				await codeReader.decodeFromVideoDevice(
+					selectedDeviceId,
+					videoElem,
+					(result, error, controls) => {
+						console.log("---- result: ", result);
```

Comment:
Are these console.logs necessary?

Reply:
Debugging purpose in dev branch it might be yes, but i think it's not really necessary too. I mean logging is very verbose without this so I could remove it for now. you can check [this](https://github.com/zxing-js/library/issues/323).

Reply:
If they are only used for debug purposes I'd say comment them out so that the user does not see them if they open the console